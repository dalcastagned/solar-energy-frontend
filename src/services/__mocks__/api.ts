import axios from 'axios';

export type Error = {
  status?: number;
  data?: Record<string, unknown>;
  message?: string;
};

export type HttpResponse<DataType> = {
  data?: DataType;
  error?: Error;
};

const api = jest.genMockFromModule('axios') as jest.MockedFunction<
  typeof axios
>;

api.create = jest.fn(() => api);

export default api;
