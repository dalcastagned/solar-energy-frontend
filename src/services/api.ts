import axios, { AxiosError, AxiosResponse } from 'axios';

// import { cleanStorage } from '../hooks/auth';

export type Error = {
  status?: number;
  data?: Record<string, unknown>;
  message?: string;
};

export type HttpResponse<DataType> = {
  data?: DataType;
  error?: Error;
};

export type ErrorResponse = {
  response: Error;
  code: string;
};

const responseSuccessHandler = (response: AxiosResponse): AxiosResponse =>
  response;

const forbidden = [401, 403];

const responseErrorHandler = (error: AxiosError): Promise<AxiosError> => {
  const isUnauthorized =
    error.response && forbidden.includes(error.response.status);
  const routers = ['/', '/reset-password'].includes(window.location.pathname);
  if (!routers && isUnauthorized) {
    // cleanStorage();
    window.location.href = '/';
  }

  return Promise.reject(error);
};

const api = axios.create({ baseURL: 'http://localhost:7140/api' });

api.interceptors.response.use(
  response => responseSuccessHandler(response),
  error => responseErrorHandler(error),
);

export default api;
