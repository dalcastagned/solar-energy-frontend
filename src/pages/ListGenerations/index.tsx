import { useCallback, useEffect, useState } from 'react';

import DeleteIcon from '@mui/icons-material/Delete';
import EditIcon from '@mui/icons-material/Edit';
import InfoIcon from '@mui/icons-material/Info';
import {
  AlertProps,
  CardContent,
  CircularProgress,
  IconButton,
  Pagination,
  Typography,
} from '@mui/material';
import Paper from '@mui/material/Paper';
import Table from '@mui/material/Table';
import TableBody from '@mui/material/TableBody';
import TableCell from '@mui/material/TableCell';
import TableContainer from '@mui/material/TableContainer';
import TableRow from '@mui/material/TableRow';
import { AdapterMoment } from '@mui/x-date-pickers/AdapterMoment';
import { DesktopDatePicker } from '@mui/x-date-pickers/DesktopDatePicker';
import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
import { useFormik } from 'formik';
import { ReadGenerations } from 'models/generations';
import moment from 'moment';
import AddGenerationModal from 'pages/modals/AddGeneration';
import UpdateGenerationModal from 'pages/modals/UpdateGeneration';
import { useLocation } from 'react-router-dom';
import { toast } from 'react-toastify';
import api from 'services/api';
import { handleErrorResponse } from 'utils/error-response';
import { notify } from 'utils/toast-notification';
import * as Yup from 'yup';

import * as S from './styles';

import 'moment/locale/pt-br';

const validationSchema = Yup.object({
  initialDate: Yup.date().nullable(),
  finalDate: Yup.date().nullable(),
});

const Cols = [
  { id: 1, name: 'ID' },
  { id: 2, name: 'Data' },
  { id: 3, name: 'Geração' },
  { id: 7, name: '' },
  { id: 8, name: '' },
];

type ShowAlertProps = AlertProps & {
  message: string;
};

const ListGenerations = (): JSX.Element => {
  const [generations, setGenerations] = useState<ReadGenerations>();
  const [page, setPage] = useState(1);
  const [loading, setLoading] = useState(false);
  const [initialDate, setInitialDate] = useState<Date | null>();
  const [finalDate, setFinalDate] = useState<Date | null>();
  const [updateGenerationId, setUpdateGenerationId] = useState(0);
  const [openAddGenerationModal, setOpenAddGenerationModal] = useState(false);
  const [openUpdateGenerationModal, setOpenUpdateGenerationModal] =
    useState(false);
  const [updateTable, setUpdateTable] = useState(false);
  const [alertProps, setAlertProps] = useState<ShowAlertProps>({
    severity: 'success',
    message: '',
  });
  const location = useLocation();

  const getGenerations = useCallback(async () => {
    setAlertProps({
      severity: 'info',
      message: '',
    });
    const response = await api.get<ReadGenerations>(
      `/plant/${location.state?.id}/generation?page=${page}&limit=8&startDate=${
        initialDate ? moment(initialDate).format('YYYY-MM-DD') : ''
      }&endDate=${finalDate ? moment(finalDate).format('YYYY-MM-DD') : ''}`,
    );
    setLoading(false);
    !response.data.generations.length &&
      setAlertProps({
        severity: 'info',
        message: 'Nenhuma geração cadastrada',
      });
    setGenerations(response.data);
    setUpdateTable(false);
  }, [page, location.state?.id, initialDate, finalDate]);

  const removeGeneration = useCallback(
    async (id: number): Promise<void> => {
      const removeToast = toast.loading('Removendo...');
      try {
        await api.delete<ReadGenerations>(
          `/plant/${location.state?.id}/generation/${id}`,
        );
        notify(removeToast, {
          message: 'Geração removida com sucesso',
          type: 'success',
        });
        getGenerations().catch(e =>
          setAlertProps({ severity: 'error', message: handleErrorResponse(e) }),
        );
      } catch (e) {
        notify(removeToast, {
          message: handleErrorResponse(e),
          type: 'error',
        });
      }
    },
    [getGenerations, location.state?.id],
  );

  const handleEdit = (id: number): void => {
    setUpdateGenerationId(id);
    setOpenUpdateGenerationModal(true);
  };

  const formik = useFormik({
    initialValues: {
      initialDate: null,
      finalDate: null,
    },
    validationSchema,
    onSubmit: values => {
      setInitialDate(values.initialDate);
      setFinalDate(values.finalDate);
    },
  });

  useEffect(() => {
    getGenerations().catch(e =>
      setAlertProps({ severity: 'error', message: handleErrorResponse(e) }),
    );
  }, [getGenerations, page, loading, updateTable]);

  return (
    <>
      {!alertProps.message.length && !generations?.generations.length ? (
        <S.Loading>
          <CircularProgress />
        </S.Loading>
      ) : (
        <S.Wrapper>
          <S.Form
            onSubmit={e => {
              e.preventDefault();
              formik.handleSubmit(e);
            }}
          >
            <LocalizationProvider dateAdapter={AdapterMoment}>
              <DesktopDatePicker
                label="Data Inicial"
                inputFormat="DD/MM/yyyy"
                value={formik.values.initialDate}
                onChange={value => {
                  formik.setFieldValue('initialDate', value);
                }}
                renderInput={params => (
                  <S.TextFieldStyled
                    {...params}
                    error={
                      formik.touched.initialDate &&
                      Boolean(formik.errors.initialDate)
                    }
                  />
                )}
              />
              <DesktopDatePicker
                label="Data Final"
                inputFormat="DD/MM/yyyy"
                value={formik.values.finalDate}
                onChange={value => {
                  formik.setFieldValue('finalDate', value);
                }}
                renderInput={params => (
                  <S.TextFieldStyled
                    {...params}
                    error={
                      formik.touched.finalDate &&
                      Boolean(formik.errors.finalDate)
                    }
                  />
                )}
              />
            </LocalizationProvider>
            <S.ButtonSearch type="submit" onClick={() => setLoading(true)}>
              {loading ? <CircularProgress color="inherit" /> : 'Pesquisar'}
            </S.ButtonSearch>
          </S.Form>
          {alertProps.message.length ? (
            <S.AlertStyled severity={alertProps.severity}>
              {alertProps.message}
            </S.AlertStyled>
          ) : null}
          {generations?.generations.length ? (
            <>
              <S.ContainerTable>
                <TableContainer component={Paper}>
                  <Table size="medium" aria-label="a dense table">
                    <S.TableHeadStyled>
                      <TableRow>
                        {Cols.map(col => (
                          <S.TableCellTopStyled key={col.id}>
                            {col.name}
                          </S.TableCellTopStyled>
                        ))}
                      </TableRow>
                    </S.TableHeadStyled>
                    <TableBody>
                      {generations?.generations.map(generation => (
                        <S.TableRowStyled key={generation.id}>
                          <S.TableCellBodyStyled width={50}>
                            {generation.id}
                          </S.TableCellBodyStyled>
                          <S.TableCellBodyStyled>
                            {moment(generation.date).format(
                              'D [de] MMMM, YYYY',
                            )}
                          </S.TableCellBodyStyled>
                          <S.TableCellBodyStyled>
                            {generation.generatePower} kw
                          </S.TableCellBodyStyled>
                          <TableCell width={80}>
                            <S.ButtonEdit
                              onClick={() => handleEdit(generation.id)}
                            >
                              Editar
                            </S.ButtonEdit>
                          </TableCell>
                          <TableCell width={110}>
                            <S.ButtonRemove
                              onClick={() => removeGeneration(generation.id)}
                            >
                              Remover
                            </S.ButtonRemove>
                          </TableCell>
                        </S.TableRowStyled>
                      ))}
                    </TableBody>
                  </Table>
                </TableContainer>
              </S.ContainerTable>
              <S.ContainerCard>
                {generations?.generations.map(generation => (
                  <S.CardStyled key={generation.id}>
                    <CardContent>
                      <S.BoxStyled>
                        <Typography variant="h4">Id:</Typography>
                        <Typography variant="h4">{generation.id}</Typography>
                      </S.BoxStyled>
                      <S.BoxStyled>
                        <Typography variant="h4">Data:</Typography>
                        <Typography variant="h4">
                          {moment(generation.date).format('D [de] MMMM, YYYY')}
                        </Typography>
                      </S.BoxStyled>
                      <S.BoxStyled>
                        <Typography variant="h4">Geração:</Typography>
                        <Typography variant="h4">
                          {generation.generatePower} kw
                        </Typography>
                      </S.BoxStyled>
                    </CardContent>
                    <S.CardActionsStyled>
                      <S.ButtonEditStyled
                        onClick={() => handleEdit(generation.id)}
                      >
                        <EditIcon />
                      </S.ButtonEditStyled>
                      <S.ButtonRemoveStyled
                        onClick={() => removeGeneration(generation.id)}
                      >
                        <DeleteIcon />
                      </S.ButtonRemoveStyled>
                    </S.CardActionsStyled>
                  </S.CardStyled>
                ))}
              </S.ContainerCard>
            </>
          ) : null}
          {generations?.generations.length ? (
            <Pagination
              count={generations?.pageCount}
              page={generations?.pageNumber}
              size="large"
              onChange={(_, value) => setPage(value)}
            />
          ) : null}
          <S.TooltipBox>
            {!location.state?.active && (
              <S.TooltipStyled
                title={
                  <span style={{ fontSize: '1.6rem' }}>
                    Não é possivel adicionar geração a uma unidade inativa
                  </span>
                }
                arrow
              >
                <IconButton>
                  <InfoIcon />
                </IconButton>
              </S.TooltipStyled>
            )}
            <S.ButtonNewGeneration
              disabled={!location.state?.active}
              onClick={() => setOpenAddGenerationModal(true)}
            >
              Nova Geração
            </S.ButtonNewGeneration>
          </S.TooltipBox>
        </S.Wrapper>
      )}
      <AddGenerationModal
        open={openAddGenerationModal}
        setOpen={setOpenAddGenerationModal}
        setUpdateTable={setUpdateTable}
      />
      <UpdateGenerationModal
        id={updateGenerationId}
        open={openUpdateGenerationModal}
        setOpen={setOpenUpdateGenerationModal}
        setUpdateTable={setUpdateTable}
      />
    </>
  );
};

export default ListGenerations;
