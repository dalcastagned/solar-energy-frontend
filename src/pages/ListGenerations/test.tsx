import { screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router-dom';
import { ToastContainer } from 'react-toastify';
import api from 'services/api';

import ListGenerations from '.';
import { renderWithRoute } from '../../utils/tests/helpers';
import {
  generationMock,
  generationsMock,
  generationsWithPageMock,
} from './__mocks__/ListGeneration';

describe('<ListGenerations />', () => {
  it('should render the main', async () => {
    jest.spyOn(api, 'get').mockResolvedValue(jest.fn());
    const { container } = renderWithRoute(<ListGenerations />);
    await waitFor(() => {
      expect(screen.getByText('Nova Geração')).toBeInTheDocument();
      expect(container).toMatchSnapshot();
    });
  });
  it('should render alert when there are no generation', async () => {
    jest.spyOn(api, 'get').mockResolvedValue({ data: { generations: [] } });

    renderWithRoute(<ListGenerations />);

    await waitFor(() => {
      expect(
        screen.queryByText('Nenhuma geração cadastrada'),
      ).toBeInTheDocument();
    });
  });
  it('should render alert when get generations fails', async () => {
    jest.spyOn(api, 'get').mockRejectedValue({ status: 500 });

    renderWithRoute(<ListGenerations />);

    await waitFor(() => {
      expect(
        screen.queryByText(
          'O servidor parece estar indisponível. Verifique sua conexão.',
        ),
      ).toBeInTheDocument();
    });
  });
  it('should remove generation desktop when click remove button', async () => {
    jest.spyOn(api, 'get').mockResolvedValue({ data: generationsMock });

    jest.spyOn(api, 'delete').mockResolvedValue(jest.fn());

    renderWithRoute(<ListGenerations />);

    await waitFor(() => {
      expect(screen.getByText('ID')).toBeInTheDocument();
    });

    jest.spyOn(api, 'get').mockRejectedValue({ status: 500 });

    userEvent.click(
      screen.getByRole('button', {
        name: 'Remover',
      }),
    );

    await waitFor(() => {
      expect(screen.getByText('ID')).toBeInTheDocument();
    });
  });
  it('should remove generation mobile when click remove button', async () => {
    jest.spyOn(api, 'get').mockResolvedValue({ data: generationsMock });

    jest.spyOn(api, 'delete').mockResolvedValue(jest.fn());

    renderWithRoute(<ListGenerations />);

    await waitFor(() => {
      expect(screen.getByText('ID')).toBeInTheDocument();
    });

    jest.spyOn(api, 'get').mockRejectedValue({ status: 500 });

    userEvent.click(screen.getByTestId('DeleteIcon'));

    await waitFor(() => {
      expect(screen.getByText('ID')).toBeInTheDocument();
    });
  });
  it('should go to page 2 when click page button', async () => {
    jest.spyOn(api, 'get').mockResolvedValue({ data: generationsWithPageMock });

    renderWithRoute(<ListGenerations />);

    await waitFor(() => {
      expect(screen.getByText('ID')).toBeInTheDocument();
    });

    jest.spyOn(api, 'get').mockResolvedValue({ data: generationsMock });

    userEvent.click(screen.getByText('2'));

    await waitFor(() => {
      expect(screen.getByText('ID')).toBeInTheDocument();
    });
  });
  it('should render error toast on remove generation error', async () => {
    jest.useFakeTimers();

    jest.spyOn(api, 'get').mockResolvedValue({ data: generationsMock });

    jest.spyOn(api, 'delete').mockRejectedValue({ status: 500 });

    renderWithRoute(
      <div>
        <ToastContainer />
        <ListGenerations />
      </div>,
    );

    await waitFor(() => {
      expect(screen.getByText('ID')).toBeInTheDocument();
    });

    userEvent.click(
      screen.getByRole('button', {
        name: 'Remover',
      }),
    );

    await waitFor(() => {
      expect(screen.queryAllByText('Removendo...')).toHaveLength(2);
    });

    await waitFor(() => {
      expect(
        screen.getByText(
          'O servidor parece estar indisponível. Verifique sua conexão.',
        ),
      ).toBeInTheDocument();
    });
  });
  it('should render add generation modal when click button and render error inputs when click to register', async () => {
    jest.spyOn(api, 'get').mockResolvedValue({ data: { generations: [] } });

    const history = createMemoryHistory();
    history.push(`/generations`, {
      id: 1,
      plantName: 'Planta 1',
      active: true,
    });

    renderWithRoute(
      <Router history={history}>
        <ListGenerations />
      </Router>,
    );

    await waitFor(() => {
      expect(
        screen.getByText('Nenhuma geração cadastrada'),
      ).toBeInTheDocument();
    });

    userEvent.click(
      screen.getByRole('button', {
        name: 'Nova Geração',
      }),
    );

    await waitFor(() => {
      screen.getByText('Cadastro de Geração');
    });

    userEvent.click(screen.getByText('Cadastrar'));

    await waitFor(() => {
      screen.getByText('A geração é obrigatória');
    });
  });
  it('should render add generation modal when click button and add new generation', async () => {
    jest.spyOn(api, 'get').mockResolvedValue({ data: { generations: [] } });
    jest.spyOn(api, 'post').mockResolvedValue(jest.fn());

    const history = createMemoryHistory();
    history.push(`/generations`, {
      id: 1,
      plantName: 'Planta 1',
      active: true,
    });

    renderWithRoute(
      <Router history={history}>
        <ListGenerations />
      </Router>,
    );

    await waitFor(() => {
      expect(
        screen.getByText('Nenhuma geração cadastrada'),
      ).toBeInTheDocument();
    });

    userEvent.click(
      screen.getByRole('button', {
        name: 'Nova Geração',
      }),
    );

    await waitFor(() => {
      screen.getByText('Cadastro de Geração');
    });

    userEvent.type(screen.getByLabelText('Data'), '04/06/2022');
    userEvent.type(screen.getByLabelText('Geração (kW)'), '150');

    userEvent.click(screen.getByText('Cadastrar'));

    await waitFor(() => {
      expect(screen.queryByText('Cadastro de Geração')).toBeNull();
    });
  });
  it('should render add generation modal when click button and render alert', async () => {
    jest.spyOn(api, 'get').mockResolvedValue({ data: { generations: [] } });
    jest.spyOn(api, 'post').mockRejectedValue({ status: 500 });

    const history = createMemoryHistory();
    history.push(`/generations`, {
      id: 1,
      plantName: 'Planta 1',
      active: true,
    });

    renderWithRoute(
      <Router history={history}>
        <ListGenerations />
      </Router>,
    );

    await waitFor(() => {
      expect(
        screen.getByText('Nenhuma geração cadastrada'),
      ).toBeInTheDocument();
    });

    userEvent.click(
      screen.getByRole('button', {
        name: 'Nova Geração',
      }),
    );

    await waitFor(() => {
      screen.getByText('Cadastro de Geração');
    });

    userEvent.type(screen.getByLabelText('Geração (kW)'), '150');

    userEvent.click(screen.getByText('Cadastrar'));

    await waitFor(() => {
      expect(
        screen.getByText('Erro! Tente novamente mais tarde.'),
      ).toBeInTheDocument();
    });
  });
  it('should render add generation modal when click button', async () => {
    jest.spyOn(api, 'get').mockResolvedValue({ data: { generations: [] } });

    const history = createMemoryHistory();
    history.push(`/generations`, {
      id: 1,
      plantName: 'Planta 1',
      active: true,
    });

    renderWithRoute(
      <Router history={history}>
        <ListGenerations />
      </Router>,
    );

    await waitFor(() => {
      expect(
        screen.getByText('Nenhuma geração cadastrada'),
      ).toBeInTheDocument();
    });

    userEvent.click(
      screen.getByRole('button', {
        name: 'Nova Geração',
      }),
    );

    await waitFor(() => {
      screen.getByText('Cadastro de Geração');
    });
  });
  it('should render update generations modal when click edit button desktop', async () => {
    jest.spyOn(api, 'get').mockResolvedValue({ data: generationsMock });

    const history = createMemoryHistory();
    history.push(`/generations`, {
      id: 1,
      plantName: 'Planta 1',
      active: true,
    });

    renderWithRoute(
      <Router history={history}>
        <ListGenerations />
      </Router>,
    );

    await waitFor(() => {
      expect(screen.getByText('ID')).toBeInTheDocument();
    });

    userEvent.click(
      screen.getByRole('button', {
        name: 'Editar',
      }),
    );

    await waitFor(() => {
      expect(screen.getByText('Edição de Geração')).toBeInTheDocument();
    });
  });
  it('should render update generation modal when click edit button and render input errors', async () => {
    jest.spyOn(api, 'get').mockResolvedValue({ data: generationsMock });

    const history = createMemoryHistory();
    history.push(`/generations`, {
      id: 1,
      plantName: 'Planta 1',
      active: true,
    });

    renderWithRoute(
      <Router history={history}>
        <ListGenerations />
      </Router>,
    );

    await waitFor(() => {
      expect(screen.getByText('ID')).toBeInTheDocument();
    });

    userEvent.click(
      screen.getByRole('button', {
        name: 'Editar',
      }),
    );

    await waitFor(() => {
      expect(screen.getByText('Edição de Geração')).toBeInTheDocument();
    });

    userEvent.click(screen.getByText('Atualizar'));

    await waitFor(() => {
      screen.getByText('A geração é obrigatória');
    });
  });
  it('should render update generation modal when click edit button and update generation', async () => {
    jest.spyOn(api, 'get').mockImplementation(url => {
      if (url === '/plant/1/generation/1') {
        return Promise.resolve({ data: generationMock });
      }
      return Promise.resolve({ data: generationsMock });
    });
    jest.spyOn(api, 'put').mockResolvedValue(jest.fn());

    const history = createMemoryHistory();
    history.push(`/generations`, {
      id: 1,
      plantName: 'Planta 1',
      active: true,
    });

    renderWithRoute(
      <Router history={history}>
        <ListGenerations />
      </Router>,
    );

    await waitFor(() => {
      expect(screen.getByText('ID')).toBeInTheDocument();
    });

    userEvent.click(
      screen.getByRole('button', {
        name: 'Editar',
      }),
    );

    await waitFor(() => {
      expect(screen.getByText('Edição de Geração')).toBeInTheDocument();
    });

    userEvent.click(screen.getByText('Atualizar'));

    await waitFor(() => {
      expect(screen.queryByText('Edição de Geração')).toBeNull();
    });
  });
  it('should render update generation modal when click edit button and render alert on error', async () => {
    jest.spyOn(api, 'get').mockImplementation(url => {
      if (url === '/plant/1/generation/1') {
        return Promise.resolve({ data: generationMock });
      }
      return Promise.resolve({ data: generationsMock });
    });
    jest.spyOn(api, 'put').mockRejectedValue({ status: 500 });

    const history = createMemoryHistory();
    history.push(`/generations`, {
      id: 1,
      plantName: 'Planta 1',
      active: true,
    });

    renderWithRoute(
      <Router history={history}>
        <ListGenerations />
      </Router>,
    );

    await waitFor(() => {
      expect(screen.getByText('ID')).toBeInTheDocument();
    });

    userEvent.click(
      screen.getByRole('button', {
        name: 'Editar',
      }),
    );

    await waitFor(() => {
      expect(screen.getByText('Edição de Geração')).toBeInTheDocument();
    });

    userEvent.click(screen.getByText('Atualizar'));

    await waitFor(() => {
      expect(
        screen.getByText('Erro! Tente novamente mais tarde.'),
      ).toBeInTheDocument();
    });
  });
  it('should render update generation modal when click edit button mobile', async () => {
    jest.spyOn(api, 'get').mockResolvedValue({ data: generationsMock });

    const history = createMemoryHistory();
    history.push(`/generations`, {
      id: 1,
      plantName: 'Planta 1',
      active: true,
    });

    renderWithRoute(
      <Router history={history}>
        <ListGenerations />
      </Router>,
    );

    await waitFor(() => {
      expect(screen.getByText('ID')).toBeInTheDocument();
    });

    userEvent.click(screen.getByTestId('EditIcon'));

    await waitFor(() => {
      expect(screen.getByText('Edição de Geração')).toBeInTheDocument();
    });
  });
  it('should search generations', async () => {
    jest.spyOn(api, 'get').mockResolvedValue({
      data: generationsMock,
    });

    const history = createMemoryHistory();
    history.push(`/generations`, {
      id: 1,
      plantName: 'Planta 1',
      active: true,
    });

    renderWithRoute(
      <Router history={history}>
        <ListGenerations />
      </Router>,
    );

    await waitFor(() => {
      expect(screen.getByText('ID')).toBeInTheDocument();
    });

    userEvent.type(screen.getByLabelText('Data Inicial'), '04/05/2022');
    userEvent.type(screen.getByLabelText('Data Final'), '05/06/2022');

    userEvent.click(screen.getByText('Pesquisar'));

    await waitFor(() => {
      expect(screen.getByText('ID')).toBeInTheDocument();
    });
  });
});
