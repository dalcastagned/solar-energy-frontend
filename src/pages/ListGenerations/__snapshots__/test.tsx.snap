// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListGenerations /> should render the main 1`] = `
<div>
  <div
    class="css-1q6wntd"
  >
    <form
      class="css-6sihv2"
    >
      <div
        class="MuiFormControl-root MuiTextField-root css-em8guj-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for=":r0:"
          id=":r0:-label"
        >
          Data Inicial
        </label>
        <div
          class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-o9k5xi-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
            id=":r0:"
            placeholder="dd/mm/yyyy"
            type="tel"
            value=""
          />
          <div
            class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
          >
            <button
              aria-label="Choose date"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="CalendarIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                />
              </svg>
            </button>
          </div>
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-1ftyaf0"
            >
              <span>
                Data Inicial
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root css-em8guj-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for=":r1:"
          id=":r1:-label"
        >
          Data Final
        </label>
        <div
          class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-o9k5xi-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
            id=":r1:"
            placeholder="dd/mm/yyyy"
            type="tel"
            value=""
          />
          <div
            class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
          >
            <button
              aria-label="Choose date"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="CalendarIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                />
              </svg>
            </button>
          </div>
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-1ftyaf0"
            >
              <span>
                Data Final
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <button
        class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-1b2e0bq-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="submit"
      >
        Pesquisar
      </button>
    </form>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardError MuiAlert-standard css-1hcet0e-MuiPaper-root-MuiAlert-root"
      role="alert"
    >
      <div
        class="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
          data-testid="ErrorOutlineIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
          />
        </svg>
      </div>
      <div
        class="MuiAlert-message css-acap47-MuiAlert-message"
      >
        O servidor parece estar indisponível. Verifique sua conexão.
      </div>
    </div>
    <div
      class="MuiBox-root css-iit57v"
    >
      <button
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1sgdkzv-MuiButtonBase-root-MuiIconButton-root"
        data-mui-internal-clone-element="true"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="InfoIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
          />
        </svg>
      </button>
      <button
        class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root Mui-disabled css-19hf11k-MuiButtonBase-root-MuiButton-root"
        disabled=""
        tabindex="-1"
        type="button"
      >
        Nova Geração
      </button>
    </div>
  </div>
</div>
`;
